external/skiboot/libstb/secvar/crypto/crypto-mbedtls.c

```diff
@@ -1,20 +1,34 @@
-// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
 /* Copyright 2021 IBM Corp.*/
-#ifdef MBEDTLS
+#ifdef SECVAR_CRYPTO_MBEDTLS
 // ^extra precaution to not compile with mbedtls unless specified
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h> // for exit
 #include "crypto.h"
-#include "include/prlog.h"
-#include "include/err.h"
+// commented out by Nick Child
+// #include <skiboot.h> // for prlog
+
+// added by Nick Child
+#include "prlog.h"
+// commented out by Nick Child
+//#include "secvar_crypto_err.h" // for err codes
+// added by Nick Child
+#include "err.h"
 
 #include <mbedtls/pk_internal.h> // for validating cert pk data
 #include <mbedtls/error.h>
 #include <mbedtls/md_internal.h>
 #include <mbedtls/oid.h>
+// Nick Child edited paths below
+// #include "libstb/crypto/pkcs7/pkcs7.h"
+// #ifdef SECVAR_CRYPTO_WRITE_FUNC
+// #include "libstb/crypto/pkcs7/pkcs7_write.h"
+// #endif
```

Comment:
Modification of code in `external`, is this necessary, or is this something we can adjust upstream as well? I know this is a bit of an old PR and I'm not up to date on where things are at the moment.

Reply:
The plan is to eventually have this code identical between skiboot and secvarctl, correct? If that is the case, then these paths should be unified between the two (somehow) so we don't need to modify skiboot code to work with secvarctl.

If there's ongoing effort elsewhere to fix this (possibly what Daniel is working on with libstb-secvar), then we can overlook this for now.